account-compression/programs/spl-compression/src/lib.rs

```diff
@@ -0,0 +1,549 @@
+//! Gummyroll is an on-chain Merkle tree that supports concurrent writes.
+//!
+//! A buffer of proof-like changelogs is stored on-chain that allow multiple proof-based writes to succeed within the same slot.
+//! This is accomplished by fast-forwarding out-of-date (or possibly invalid) proofs based on information stored in the changelogs.
+//! See a copy of the whitepaper [here](https://google.com)
```

Comment:
Let's fix these comments for when the crate gets published

Reply:
Updated

Reply:
Actually this whole paragraph needs to be rewritten